File: Section 1 Introduction/programming-challenges/maximum_pairwise_product.py
def maximum_pairwise_product(numbers):
    max1 = max2 = 0

    for num in numbers:
        if num > max1:
            max2 = max1
            max1 = num
        elif num > max2:
            max2 = num

    return max1 * max2


n = int(input("Enter number of elements: "))
x = list(map(int, input("Enter the numbers separated by space: ").split()))

if len(x) != n:
    print("Error: Number of elements doesn't match.")
else:
    print("Maximum pairwise product is:", maximum_pairwise_product(x))

File: Section 1 Introduction/programming-challenges/maximum_pairwise_product.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter numbers separated by spaces: ", function(input) {
    const numbers = input.split(" ").map(Number);

    if (numbers.length < 2) {
        console.log("Please enter at least two numbers.");
      } else {
        let max1 = -Infinity;
        let max2 = -Infinity;
    
        for (let num of numbers) {
          if (num > max1) {
            max2 = max1;
            max1 = num;
          } else if (num > max2) {
            max2 = num;
          }
        }
    
        const product = max1 * max2;
        console.log("Maximum pairwise product is:", product);
      }
    
      rl.close();
})